tmail-backend/webadmin/webadmin-rate-limit/src/main/java/com/linagora/tmail/webadmin/RateLimitPlanUserRoutes.java

```diff
@@ -92,13 +90,6 @@ private Route attachPlanToUser() {
             planIdPreconditions(planId);
             return Mono.from(planUserRepository.applyPlan(username, planId))
                 .then(Mono.just(Responses.returnNoContent(response)))
-                .onErrorResume(NullPointerException.class, e -> {
-                    throw ErrorResponder.builder()
-                        .statusCode(HttpStatus.BAD_REQUEST_400)
-                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
-                        .message(e.getMessage())
-                        .haltError();
-                })
```

Comment:
I'am just curious.... Why did you wrote this in the first place?

It looks like dead code that was never called....

Reply:
In repository API we have a Precondition check username/planId null. I was not sure the code parsing these in James handles all exception cases and maybe returns null.
I think this is overthinking...

Reply:
TDD is the solution to overthinking and to writing the code you truely need,